packages/legacy-events/ReactSyntheticEventType.js

```diff
@@ -31,8 +31,8 @@ export type ReactSyntheticEvent = {|
     nativeEventTarget: EventTarget,
   ) => ReactSyntheticEvent,
   isPersistent: () => boolean,
-  _dispatchInstances: null | Array<Fiber>,
-  _dispatchListeners: null | Array<Function>,
-  _targetInst: null | Fiber,
+  _dispatchInstances: null | Array<Fiber> | Fiber,
+  _dispatchListeners: null | Array<Function> | Function,
```

Comment:
This type looks sketchy/unfortunate.

Reply:
It was a genuine fix. `_dispatchInstances`/`_dispatchListeners` can be a single node rather than array. We don't excipit it with the recent changes we made, but legacy `accumulate*` try to be clever in not allocating arrays where possible so the Flow type accounts for that now.

Reply:
Yeah I noticed that about the accumulate method. Just seems like a very overloaded type, easy to mess up 😅

Reply:
Yeah, that's why I'm making these refactors. The sooner we move away from that style, the better. :)

Reply:
Fair enough!